go/test/endtoend/orchestrator/orc_test.go

```diff
@@ -107,101 +126,243 @@ func createCluster(t *testing.T, numReplicas int, numRdonly int) *cluster.LocalP
 func TestMasterElection(t *testing.T) {
 	defer cluster.PanicHandler(t)
 	clusterInstance := createCluster(t, 1, 1)
+	keyspace := &clusterInstance.Keyspaces[0]
+	shard0 := &keyspace.Shards[0]
 	defer func() {
 		clusterInstance.Teardown()
+		killTablets(t, shard0)
 	}()
+
+	//log.Exitf("error")
+	checkMasterTablet(t, clusterInstance, shard0.Vttablets[0])
+	checkReplication(t, clusterInstance, shard0.Vttablets[0], shard0.Vttablets[1:])
+}
+
+// 2. bring down master, let orc promote replica
+func TestDownMaster(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	clusterInstance := createCluster(t, 2, 0)
 	keyspace := &clusterInstance.Keyspaces[0]
 	shard0 := &keyspace.Shards[0]
-	for _, tablet := range shard0.Vttablets {
-		// Reset status, don't wait for the tablet status. We will check it later
-		tablet.VttabletProcess.ServingStatus = ""
+	defer func() {
+		clusterInstance.Teardown()
+		killTablets(t, shard0)
+	}()
+	// find master from topo
+	curMaster := shardMasterTablet(t, clusterInstance, keyspace, shard0)
+	assert.NotNil(t, curMaster, "should have elected a master")
 
-		// Start the tablet
-		err := tablet.VttabletProcess.Setup()
-		require.NoError(t, err)
+	// Make the current master database unavailable.
+	err := curMaster.MysqlctlProcess.Stop()
+	require.NoError(t, err)
+
+	for _, tablet := range shard0.Vttablets {
```

Comment:
where's the `Sleep()` now?

Reply:
It is inside the call to checkMasterTablet